### PR TITLE
Add `random_query` example

### PR DESCRIPTION
**Pull request by mfdorst**
# Objective

Showcase how to select random elements from a query, as requested in #3652.

## Solution

Add an example that shows how one can use [`rand::seq::IteratorRandom::choose`] and [`rand::seq::IteratorRandom::choose_multiple`] to choose one or multiple random items from a query.

[`rand::seq::IteratorRandom::choose`]: https://docs.rs/rand/latest/rand/seq/trait.IteratorRandom.html#method.choose
[`rand::seq::IteratorRandom::choose_multiple`]: https://docs.rs/rand/latest/rand/seq/trait.IteratorRandom.html#method.choose_multiple

### CODE REVIEW

File: examples/ecs/random_query.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+fn generate_names(mut commands: Commands) {
+    let names = ["Linus", "Noah", "Elijah", "Olivia", "Ava", "Charlotte"];
+    for name in names {
+        commands.spawn().insert(Name(name.to_owned()));
+    }
+}
+
+fn generate_pets(mut commands: Commands) {
+    let pets = ['🐀', '🐄', '🐅', '🐇', '🐊', '🐓', '🐖'];
```
Comment (alice-i-cecile) [inline examples/ecs/random_query.rs:26]: 🔥

File: examples/README.md [hunk 2]
```diff
@@ -168,6 +168,7 @@ Example | File | Description
 `hierarchy` | [`ecs/hierarchy.rs`](./ecs/hierarchy.rs) | Creates a hierarchy of parents and children entities
 `iter_combinations` | [`ecs/iter_combinations.rs`](./ecs/iter_combinations.rs) | Shows how to iterate over combinations of query results.
 `parallel_query` | [`ecs/parallel_query.rs`](./ecs/parallel_query.rs) | Illustrates parallel queries with `ParallelIterator`
+`random_query` | [`ecs/random_query.rs`](./ecs/random_query.rs) | Get random components from a query with `rand::seq::IteratorRandom`
```
Comment (alice-i-cecile) [inline examples/README.md:171]:
```suggestion
`random_query` | [`ecs/random_query.rs`](./ecs/random_query.rs) | Sample random elements of a query with `rand::seq::IteratorRandom`
```
The returned items are items of the iterator, not individual components

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Small description nit, but this example is *excellent*. Very clear, and it's a very handy trick. Thanks!

File: examples/README.md (hunk 2 above)
Comment (mfdorst) [reply examples/README.md:171]: Ah thank you. I went back and forth on how to word that.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by IceSentry** [comment]
There doesn't seem to be anything bevy specific about this example, this is only showing how to use the IteratorRandom trait from rand. Do we really need an example for that?

**Comment by alice-i-cecile** [comment]
Point taken. We can integrate this into a game example as the need arises. Closing down this and the originating issue as this was much easier than I expected at first glance.

**Pull request closed without merging**